packages/vitest/src/integrations/chai/jest-expect.ts

```diff
@@ -122,10 +122,11 @@ export const JestChaiExpect: ChaiPlugin = (chai, utils) => {
     )
   })
   def('toMatch', function(expected: string | RegExp) {
-    if (typeof expected === 'string')
-      return this.include(expected)
-    else
-      return this.match(expected)
+    if (typeof expected === 'string') { return this.include(expected) }
+    else {
+      const received = this._obj
+      return new RegExp(expected).test(received)
```

Comment:
returning doesn't actually do anything 👀 
you need to use `this.assert`
also a comment would be nice to have explaining the decision